src/Reflex/Dom/Builder/Class.hs

```diff
@@ -462,14 +462,20 @@ instance (DomBuilder t m, Monoid w, MonadHold t m, MonadFix m) => DomBuilder t (
 
 -- * Convenience functions
 
+class Unconstrained g h
+instance Unconstrained g h
+
+class DomBuilderSpace a ~ DomBuilderSpace b => SameDomBuilderSpace a b
+instance DomBuilderSpace a ~ DomBuilderSpace b => SameDomBuilderSpace a b
```

Comment:
For GHC 7.10 and above, we should ideally mark these instances `{-# INCOHERENT #-}`. They're not, of course; the classes have no methods. But GHC won't reduce them eagerly to their constraints because someone, somewhere, could (for no possible reason) write instances that overlap.